src/types.ts

```diff
@@ -770,6 +807,37 @@ function isValidCidr(ip: string, version?: IpVersion) {
   return false;
 }
 
+function isValidCreditCard(cardNumber: string): boolean {
+  const isLuhnAlgo = (cardNumber: string): boolean => {
+    // Remove any non-digit chars
+    const number = cardNumber.replace(nonDigitRegex, "");
+
+    // Create necessary variables
+    let length = number.length;
+    let bit = 1;
+    let sum = 0;
+
+    // Calculate sum of algorithm
+    while (length) {
+      const value = +number[--length];
+      bit ^= 1;
+      sum += bit ? [0, 2, 4, 6, 8, 1, 3, 5, 7, 9][value] : value;
+    }
+
+    // Return whether its valid
+    return sum % 10 === 0;
+  };
+
+  const sanitized = cardNumber.replace(creditSanitizeRegex, "");
+  return (creditCardRegex.test(cardNumber) &&
+    // Remove any hyphens and blanks
+    sanitized &&
+    // Check if it matches a provider
+    creditByProviderRegexList.some((regex) => regex.test(sanitized!)) &&
+    // Check if passes luhn algorithm
+    isLuhnAlgo(sanitized)) as boolean;
```

Comment:
already typed as a function returning `boolean`

```suggestion
    isLuhnAlgo(sanitized));
```